service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -521,21 +526,26 @@ public ReportWithLibraryInProgress createReportWithLibraryInProgress(JWTData use
 
     SchoolReportsInProgressLibrariesRecord newReport =
         db.newRecord(SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES);
-    newReport.setApprenticeshipProgram(ApprenticeshipProgram.
-        from(upsertRequest.getApprenticeshipProgram()));
-    newReport.setAssignedPersonRole(TimeRole.from(upsertRequest.getAssignedPersonRole()));
-    newReport.setHasInvitingSpace(upsertRequest.getHasInvitingSpace());
-    newReport.setIsSharedSpace(upsertRequest.getSharedSpace());
-    newReport.setMostRecentShipmentYear(upsertRequest.getMostRecentShipmentYear());
+
+    newReport.setSchoolId(schoolId);
     newReport.setUserId(userData.getUserId());
-    newReport.setNumberOfBooks(upsertRequest.getNumberOfBooks());
     newReport.setNumberOfChildren(upsertRequest.getNumberOfChildren());
-    newReport.setAssignedPersonTitle(ApprenticeTitle.from(upsertRequest.getAssignedPersonTitle()));
+    newReport.setNumberOfBooks(upsertRequest.getNumberOfBooks());
+    newReport.setMostRecentShipmentYear(upsertRequest.getMostRecentShipmentYear());
+    newReport.setApprenticeshipProgram(upsertRequest.getApprenticeshipProgram());
+    newReport.setHasInvitingSpace(upsertRequest.getHasInvitingSpace());
+    newReport.setIsSharedSpace(upsertRequest.getIsSharedSpace());
+    newReport.setAssignedPersonRole(upsertRequest.getAssignedPersonRole());
+    newReport.setAssignedPersonTitle(upsertRequest.getAssignedPersonTitle());
     newReport.setTrainsAndMentorsApprentices(upsertRequest.getTrainsAndMentorsApprentices());
+
     newReport.store();
+    newReport.refresh();
```

Comment:
What does the refresh do and does store() need to be called beforehand?

Reply:
`refresh()` forces jooq to make a query to the database to grab the actual record and update the object (in this case, update `newReport`).

`store()` inserts the record, but doesn't update the `newReport` object to contain the actual values in the DB. I added a `refresh()` here mainly to grab the `created_at` and `updated_at` timestamps, since we are sending those back in our responses, but it's also a good way to ensure we're sending accurate information back in the response.